File: 단계별로 풀어보기/Step 10/1978.cpp
#include <iostream>
using namespace std;

int main() {
    int N;
    cin >> N;

    int i, num;
    int cnt = 0;
    for(i=0; i<N; i++) {
        cin >> num;
        if(num==2) cnt++;
        else if(num>2){
            int j;
            for(j=2; j<num; j++) {
                if(num%j == 0) break;
            }
            if(j==num) cnt++;
        }
    }

    cout << cnt;
    return 0;
}

File: StepByStep/Python/Step_08_기본 수학 2/1978.py
import math
N = int(input())
nums = map(int, input().split())
cnt = 0
for num in nums:
    if num == 2:
        cnt += 1
    elif num > 2:
        prime = True
        for i in range(2, int(math.sqrt(num)) + 1):
            if num % i == 0:
                prime = False
                break
        if prime:
            cnt += 1
print(cnt)